superset-frontend/src/dashboard/components/nativeFilters/FilterBar/utils.ts

```diff
@@ -39,3 +40,9 @@ export function mapParentFiltersToChildren(
   });
   return cascadeChildren;
 }
+
+export const getOnlyExtraFormData = (data: DataMaskStateWithId) =>
+  Object.values(data).reduce(
+    (prev, next) => ({ ...prev, [next.id]: next.filterState }),
```

Comment:
I believe this should be
```suggestion
    (prev, next) => ({ ...prev, [next.id]: next.extraFormData }),
```